### PR TITLE
Updated data check code for regression

### PR DESCRIPTION
**Pull request by simha104**
### Updated data check code for regression
This is to avoid duplicate errors produced by non numerical regression and unsupported data types

### CODE REVIEW

File: evalml/data_checks/data_check_message_code.py [hunk 1]
```diff
@@ -32,6 +32,9 @@ class DataCheckMessageCode(Enum):
     TARGET_UNSUPPORTED_TYPE = "target_unsupported_type"
     """Message code for target data that is of an unsupported type."""
 
+    TARGET_REGRESSION_NON_NUMERIC = "target_regression_non_numeric"
```
Comment (chukarsten) [inline evalml/data_checks/data_check_message_code.py:35]: Maybe we just rename this to `TARGET_UNSUPPORTED_TYPE_REGRESSION`?  As named, it doesn't seem to get the point across to me.  Also a nit: "incompatible with regression"

**Review by chukarsten — CHANGES_REQUESTED** [review]
Thanks for making the adjustment!  I think we should still adjust the name, though.

**Review by bchen1116 — APPROVED** [review]
Agreed with Karsten here, but looks good otherwise.

**Review by chukarsten — APPROVED** [review]
Thanks a bunch!

**Pull request merged**